src/Npgsql/Internal/NpgsqlDatabaseInfo.cs

```diff
@@ -134,11 +142,28 @@ public abstract class NpgsqlDatabaseInfo
         /// Initializes the instance of <see cref="NpgsqlDatabaseInfo"/>.
         /// </summary>
         protected NpgsqlDatabaseInfo(string host, int port, string databaseName, Version version)
+            : this(host, port, databaseName, version, version.ToString())
+        { }
+
+        /// <summary>
+        /// Initializes the instance of <see cref="NpgsqlDatabaseInfo"/>.
+        /// </summary>
+        protected NpgsqlDatabaseInfo(string host, int port, string databaseName, Version version, string serverVersion)
```

Comment:
Well, they are protected constructors in a public abstract class, so in a way they are public API.
That's why I kept the first one for backwards compatibility reasons, although it is kind of defunct now because it sets the `ServerVersion` string to `Version.ToString()` (as we always did before this PR).

I did this because I have no idea which hypothetical databse API (Redshift, Greenplum, whatever...) might inherit from `NpgsqlDatabaseInfo` and how feasible it would be for them to add the `serverVersion` argument.

The second one is intended as the new public API with full functionality (including the full `serverVersion`) string. It is meant as natural extension of the first one (just add the `serverVersion` argument).

The third one is the one actually used from the `PostgresDatabaseInfo` constructor.

I don't really care about them but removing the first one would be an avoidable breaking change on a public API and omitting the second one would leave the ServerVersion feature unsupported on the public API.

Reply:
> Well, they are protected constructors in a public abstract class, so in a way they are public API.

That's true, though note that at least for now this class is pubternals. Am not 100% sure that's best but maybe we should wait for actual requests to extend this etc.

I guess I'm OK with leaving things like this... I highly doubt there will be a need to write an NpgsqlDataInfo that cannot provide a string version, but there's little downside...